### PR TITLE
refactor: Startup

### LINKED ISSUE #12: Refactor Startup to use DI appropriately
The current Startup is not taking advantage of the declarative nature of Dotnet DI.

### PR DESCRIPTION
**Pull request by tracyboehrer**
Fixes #12 

Initial Startup refactor to utilize declarative DI better.

I don't think this is done.

1) There are some warts concerning BotSettings vs IConfiguration.  SDK, specifically BotFrameworkHttpAdapter, need an IConfiguration.  But it sometimes easier to use the bound BotSettings, and I don't think it's clean to pass both around.

### CODE REVIEW

File: CoreBot/runtime/core/RuntimeAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.BotFramework.Composer.Core.Settings;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.BotFramework.Composer.Core
+{
+    public class RuntimeAdapter : BotFrameworkHttpAdapter
+    {
+        public RuntimeAdapter(
+                IConfiguration configuration,
```
Comment (peterinnesmsft) [inline CoreBot/runtime/core/RuntimeAdapter.cs:17]: nit: untab by one

File: CoreBot/runtime/core/RuntimeAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                IConfiguration configuration, 
+                IStorage storage, 
+                UserState userState, 
+                ConversationState conversationState, 
+                TelemetryInitializerMiddleware telemetryInitializerMiddleware)
+            : base(configuration)
+        {
+            this.UseStorage(storage);
+            this.UseBotState(userState, conversationState);
+            Use(new RegisterClassMiddleware<IConfiguration>(configuration));
```
Comment (peterinnesmsft) [inline CoreBot/runtime/core/RuntimeAdapter.cs:26]: Not that there's anything you can do to change this, but I hate that this function isn't named UseMiddleware... :'(

File: CoreBot/runtime/core/Settings/BotSettings.cs [hunk 3]
```diff
@@ -37,5 +37,10 @@ public class AdditionalTelemetryConfiguration
 
             public bool LogActivities { get; set; }
         }
+
+        public static bool ConfigSectionValid(string val)
+        {
+            return !string.IsNullOrEmpty(val) && !val.StartsWith("<");
```
Comment (peterinnesmsft) [inline CoreBot/runtime/core/Settings/BotSettings.cs:43]: The IsNullOrEmpty makes sense...not sure I follow the '<' though?

Comment (tracyboehrer) [reply CoreBot/runtime/core/Settings/BotSettings.cs:43]: I don't know either.  I had the same question.  This was moved from the original Startup.cs.

Comment (peterinnesmsft) [reply CoreBot/runtime/core/Settings/BotSettings.cs:43]: 🙃🙃🙃

File: CoreBot/runtime/core/RuntimeAdapter.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+            Use(telemetryInitializerMiddleware);
```
Comment (peterinnesmsft) [inline CoreBot/runtime/core/RuntimeAdapter.cs:28]: >Use [](start = 12, length = 3)

Another nit: looks like this is a member function, and above we're using this.UseStorage, this.UseBotState, etc., so I guess we should be consistent.

I'm not sure what the broader team stance is. My personal preference is to use this. as I think it helps greatly with readability...but whatever ultimately aligns us.

Comment (tracyboehrer) [reply CoreBot/runtime/core/RuntimeAdapter.cs:28]: So this is probably a shortcoming on my part.  "UseBotState" is an extension, and not using "this." wouldn't work.  Is there another way?

File: CoreBot/runtime/core/ComposerBot.cs [hunk 5]
```diff
@@ -29,16 +33,29 @@ public class ComposerBot : ActivityHandler
...
             this.dialogState = conversationState.CreateProperty<DialogState>("DialogState");
             this.resourceExplorer = resourceExplorer;
-            this.rootDialogFile = rootDialog;
-            this.defaultLocale = defaultLocale;
+            this.defaultLocale = configuration.GetValue<string>("defaultLanguage") ?? "en-us"; ;
             this.telemetryClient = telemetryClient;
-            this.removeRecipientMention = removeRecipientMention;
+
+            // TODO: Could this not be a setting (as opposed to a hard coded location)?
+            this.rootDialogFile = GetRootDialog(configuration["bot"]);
```
Comment (ryanlengel) [inline CoreBot/runtime/core/ComposerBot.cs:53]: Yes this should be a setting, this is provided in the GetComposerPathExtensions method in the Core project and should be decoupled.

**Pull request closed without merging**